src/main.rs

```diff
@@ -50,6 +50,41 @@ fn handle_addition(settings: &Settings) {
 
 fn handle_search(settings: &Settings) {
     let history = History::load(settings.history_format);
+
+    if settings.no_ui {
+        history.build_cache_table(
+            &settings.dir.to_owned(),
+            &Some(settings.session_id.to_owned()),
+            None,
+            None,
+            None,
+            settings.limit.to_owned(),
+        );
+        let matches = history.find_matches(
+            &settings.command,
+            settings.results as i16,
+            settings.fuzzy,
+            &mcfly::settings::ResultSort::Rank,
+        );
+
+        let cmd = match matches.first() {
+            Some(cmd) => cmd.cmd.to_owned(),
+            None => settings.command.to_owned(),
```

Comment:
It seems like if it cannot find a match, it should return nothing, not the search query?

Reply:
Yes, I somehow misunderstood what the original UI code was doing.

Reply:
This is still returning the search query, right?